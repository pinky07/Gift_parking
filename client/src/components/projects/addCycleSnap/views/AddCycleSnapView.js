import React from 'react';

import Heading from 'grommet/components/Heading';
import Form from 'grommet/components/Form';
import FormFields from 'grommet/components/FormFields';
import FormField from 'grommet/components/FormField';
import TextInput from 'grommet/components/TextInput';
import Box from 'grommet/components/Box';
import Footer from 'grommet/components/Footer';
import Button from 'grommet/components/Button';
import NumberInput from 'grommet/components/NumberInput';
import DateTime from 'grommet/components/DateTime';
import Layer from 'grommet/components/Layer';
import CheckBox from 'grommet/components/CheckBox';

const AddCycleSnapView = ({ viewModel }) => {
  const { errors } = viewModel;
  const { formCallbacks } = viewModel;
  const { projectName } = viewModel;
  const { cycleSnapName } = viewModel;
  const { startDate } = viewModel;
  const { endDate } = viewModel;
  const { targetedPoints } = viewModel;
  const { achievedPoints } = viewModel;
  const { isMoodAvailable } = viewModel;
  const { mood } = viewModel;
  const { isWasteAvailable } = viewModel;
  const { teamCapacity } = viewModel;
  const { wasteDays } = viewModel;

  let moodAverageFormField;
  if (isMoodAvailable)
    moodAverageFormField = <FormField label='Mood Average' htmlFor='moodId' error={errors.mood}>
      <NumberInput id='moodId' step={0.01} min={1} max={3} value={mood} onChange={formCallbacks.onMoodChange} />
    </FormField>;

  let teamCapacityFormField;
  let wasteDaysFormField;
  if (isWasteAvailable) {
    teamCapacityFormField = <FormField label='Team capacity' htmlFor='teamCapacityId' error={errors.teamCapacity}>
      <NumberInput id='teamCapacityId' step={0.01} min={1} max={10000} value={teamCapacity} onChange={formCallbacks.onTeamCapacityChange} />
    </FormField>;
    wasteDaysFormField = <FormField label='Waste days' htmlFor='wasteDaysId' error={errors.wasteDays}>
      <NumberInput id='wasteDaysId' step={0.01} min={0} max={10000} value={wasteDays} onChange={formCallbacks.onWasteDaysChange} />
    </FormField>;
  }

  return (<Layer align='center' closer={true} onClose={formCallbacks.onClose}>
    <Box pad={{ vertical: 'large', horizontal: 'small' }}>

      <Heading>Add cycle snap</Heading>
      <h2>Project: {projectName}</h2>

      <Form compact={false}>
        <FormFields>
          <fieldset>
            <FormField label='Name' htmlFor="cycleNameInput" error={errors.name}>
              <TextInput id='cycleNameInput' onDOMChange={formCallbacks.onNameChange} value={cycleSnapName} />
            </FormField>

            <FormField label='Start Date' htmlFor="cycleStartDate" error={errors.startDate}>
              <DateTime id='cycleStartDate' format='YYYY-MM-DD' value={startDate} onChange={formCallbacks.onStartDateChange} />
            </FormField>

            <FormField label='End Date' htmlFor="cycleEndDate" error={errors.endDate}>
              <DateTime id='cycleEndDate' format='YYYY-MM-DD' value={endDate} onChange={formCallbacks.onEndDateChange} />
            </FormField>

            <FormField label='Targeted points' htmlFor='targetedPointsId' error={errors.targetedPoints}>
              <NumberInput id='targetedPointsId' step={1} value={targetedPoints} min={0} onChange={formCallbacks.onTargetedPointsChange} />
            </FormField>

            <FormField label='Achieved points' htmlFor='achievedPointsId' error={errors.achievedPoints}>
              <NumberInput id='achievedPoints' step={1} value={achievedPoints} min={0} onChange={formCallbacks.onAchievedPointsChange} />
            </FormField>

            <FormField>
              <CheckBox label='Is mood available?'
                reverse={true}
                toggle={true}
                defaultChecked={true}
                onChange={formCallbacks.onIsMoodAvailableChange} />
            </FormField>

            {moodAverageFormField}

            <FormField>
              <CheckBox label='Is waste available?'
                reverse={true}
                toggle={true}
                defaultChecked={true}
                onChange={formCallbacks.onIsWasteAvailableChange} />
            </FormField>

            {teamCapacityFormField}
            {wasteDaysFormField}

          </fieldset>
        </FormFields>

        <Footer pad={{ "vertical": "medium" }}>
          <Box>
            <Button label='Add' type='submit' primary={true} onClick={formCallbacks.onClick} />
          </Box>
          <Box pad={{ horizontal: 'small' }}>
            <Button label='Cancel' secondary={true} onClick={formCallbacks.onClose} />
          </Box>
        </Footer>
      </Form>
    </Box>
  </Layer>);
}

export default AddCycleSnapView;