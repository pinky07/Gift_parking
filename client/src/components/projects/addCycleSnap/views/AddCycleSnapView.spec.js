import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import AddCycleSnapView from './AddCycleSnapView';

describe('AddCycleSnapView', () => {

  it('should render the correct data values and validation errors', () => {
    let viewModel = getViewModelWithErrors();

    const renderer = ShallowRenderer.createRenderer();
    const actual = renderer.render(<AddCycleSnapView viewModel={viewModel}/>);

    expect(actual).toMatchSnapshot();
  });

  function getViewModelWithErrors() {
    return {
      projectId: '7',
      projectName: 'Exceptional Project',
      cycleSnapName: '1',
      startDate: '2017-01-02',
      endDate: '2017-01-15',
      targetedPoints: '8',
      achievedPoints: '13',
      errors: {
        name: 'error name',
        startDate: 'error startDate',
        endDate: 'error endDate',
        targetedPoints: 'error targetedPoints',
        achievedPoints: 'error achievedPoints'
      },
      formCallbacks: {
        onNameChange: jest.fn(),
        onStartDateChange: jest.fn(),
        onEndDateChange: jest.fn(),
        onTargetedPointsChange: jest.fn(),
        onAchievedPointsChange: jest.fn(),
        onClick: jest.fn(),
        onClose: jest.fn()
      }
    };
  }
});