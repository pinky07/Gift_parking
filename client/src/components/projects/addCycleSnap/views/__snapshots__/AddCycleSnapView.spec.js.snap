// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCycleSnapView should render the correct data values and validation errors 1`] = `
<Layer
  align="center"
  closer={true}
  onClose={[Function]}
>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad={
      Object {
        "horizontal": "small",
        "vertical": "large",
      }
    }
    responsive={true}
    tag="div"
  >
    <Heading
      tag="h1"
    >
      Add cycle snap
    </Heading>
    <h2>
      Project: 
      Exceptional Project
    </h2>
    <Form
      compact={false}
      fill={false}
      pad="none"
    >
      <FormFields>
        <fieldset>
          <FormField
            error="error name"
            htmlFor="cycleNameInput"
            label="Name"
            size="medium"
            strong={false}
          >
            <TextInput
              id="cycleNameInput"
              onDOMChange={[Function]}
              value="1"
            />
          </FormField>
          <FormField
            error="error startDate"
            htmlFor="cycleStartDate"
            label="Start Date"
            size="medium"
            strong={false}
          >
            <DateTime
              format="YYYY-MM-DD"
              id="cycleStartDate"
              onChange={[Function]}
              step={1}
              value="2017-01-02"
            />
          </FormField>
          <FormField
            error="error endDate"
            htmlFor="cycleEndDate"
            label="End Date"
            size="medium"
            strong={false}
          >
            <DateTime
              format="YYYY-MM-DD"
              id="cycleEndDate"
              onChange={[Function]}
              step={1}
              value="2017-01-15"
            />
          </FormField>
          <FormField
            error="error targetedPoints"
            htmlFor="targetedPointsId"
            label="Targeted points"
            size="medium"
            strong={false}
          >
            <NumberInput
              id="targetedPointsId"
              min={0}
              onChange={[Function]}
              step={1}
              value="8"
            />
          </FormField>
          <FormField
            error="error achievedPoints"
            htmlFor="achievedPointsId"
            label="Achieved points"
            size="medium"
            strong={false}
          >
            <NumberInput
              id="achievedPoints"
              min={0}
              onChange={[Function]}
              step={1}
              value="13"
            />
          </FormField>
          <FormField
            size="medium"
            strong={false}
          >
            <CheckBox
              defaultChecked={true}
              label="Is mood available?"
              onChange={undefined}
              reverse={true}
              toggle={true}
            />
          </FormField>
          <FormField
            size="medium"
            strong={false}
          >
            <CheckBox
              defaultChecked={true}
              label="Is waste available?"
              onChange={undefined}
              reverse={true}
              toggle={true}
            />
          </FormField>
        </fieldset>
      </FormFields>
      <Footer
        align="center"
        direction="row"
        pad={
          Object {
            "vertical": "medium",
          }
        }
        responsive={false}
      >
        <Box
          announce={false}
          direction="column"
          focusable={true}
          pad="none"
          responsive={true}
          tag="div"
        >
          <Button
            label="Add"
            method="push"
            onClick={[Function]}
            primary={true}
            type="submit"
          />
        </Box>
        <Box
          announce={false}
          direction="column"
          focusable={true}
          pad={
            Object {
              "horizontal": "small",
            }
          }
          responsive={true}
          tag="div"
        >
          <Button
            label="Cancel"
            method="push"
            onClick={[Function]}
            secondary={true}
            type="button"
          />
        </Box>
      </Footer>
    </Form>
  </Box>
</Layer>
`;
